CleverPush/CleverPush/Source/CPAppBannerModuleInstance.m

```diff
@@ -268,9 +268,50 @@ - (BOOL)bannerTargetingAllowed:(CPAppBanner*)banner {
         }
     }
 
+    allowed = allowed ? [self appVersionFilter:banner] : NO;
     return allowed;
 }
 
+#pragma mark - check the banner triggering allowed as per selected version match with app version or not.
+-(BOOL)appVersionFilter:(CPAppBanner*)banner{
+    if (banner.appVersionFilterValue == nil || [banner.appVersionFilterValue isEqualToString:@""]) {
+        return NO;
```

Comment:
IF `appVersionFilterValue` is nil or empty string, allowed would always be false. This is not what we want.

We just want to set `allowed` to false if any of the conditions below does NOT apply.